In [18]:
# Import packages
import pandas as pd
import os
import numpy as np
import xarray as xr
import rioxarray as rioxr
import geopandas as gpd

pd.set_option('display.max_columns', None) 

In [19]:
# Import fire data

ca_fire = gpd.read_file(os.path.join('data', 
                                    'California_Fire_Perimeters_(all).shp'))

In [21]:
# Explore first 5 values
ca_fire.head()

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
0   2023    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17  2023-06-17   
1   2023    CA    LRA     BTU     KAISER  00010225  2023-06-02  2023-06-02   
2   2023    CA    CDF     AEU    JACKSON  00017640  2023-07-01  2023-07-02   
3   2023    CA    CDF     AEU     CARBON  00018821  2023-07-11  2023-07-11   
4   2023    CA    CDF     AEU    LIBERTY  00018876  2023-07-11  2023-07-12   

   CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0      5         1          1    5.72913     None       None   
1      5         1          1   13.60240     None       None   
2      2         1          1   27.81450     None       None   
3      9         1          1   58.76020     None       None   
4     14         1          1   70.97900     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None     2020   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None     2020   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None     2020   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}     None       None     2020   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}     None       None     2020   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...  
3  POLYGON ((-13468077.000 4642260.000, -13467975...  
4  POLYGON ((-13468418.000 4614853.000, -13468428...

In [22]:
# View data types
ca_fire.dtypes

YEAR_            int64
STATE           object
AGENCY          object
UNIT_ID         object
FIRE_NAME       object
INC_NUM         object
ALARM_DATE      object
CONT_DATE       object
CAUSE            int64
C_METHOD         int64
OBJECTIVE        int64
GIS_ACRES      float64
COMMENTS        object
COMPLEX_NA      object
IRWINID         object
FIRE_NUM        object
COMPLEX_ID      object
DECADES          int64
geometry      geometry
dtype: object

In [24]:
# View size of entire df
ca_fire.shape

(22261, 19)

## Preliminary Observations

From my preliminary exploration, I found that the CA_fire dataframe contains fire data with an integer year type, with other columns as mostly string data types. Alternatively, the cause, method, and objective of the fires are classified as integers. There are not many NA values, however it is noted in the comments, for example, if no comments were made. Looking at the shape, there are 22,261 fire observations with 19 classifying columns

In [30]:
# Subset to Thomas Fire in 2017
thomas_fire = ca_fire[(ca_fire.FIRE_NAME == 'THOMAS') &
                     (ca_fire.YEAR_ == 2017)]

In [33]:
# Save Thomas Fire to shapefile in data folder
thomas_fire.to_file('data/thomas_fire.shp')

I selected this shapefile type because it allows us the correct geometries for plotting against the landsat data in the next part. This allows us to plot more easily.